projects/forest_firefighters/controllers/autonomous_mavic/autonomous_mavic.py

```diff
@@ -0,0 +1,285 @@
+from controller import Robot
+import sys, random, optparse
+try:
+    import numpy as np
+    from numpy import NaN, nan
+except ImportError:
+    sys.exit("Warning: 'numpy' module not found. Please check the Python modules installation instructions " +
+             "at 'https://www.cyberbotics.com/doc/guide/using-python'.")
+try:
+    import cv2
+except ImportError:
+    sys.exit("Warning: 'cv2' module not found. Please check the Python modules installation instructions " +
+             "at 'https://www.cyberbotics.com/doc/guide/using-python'.")
+
+def clamp(value, value_min, value_max):
+    return min(max(value, value_min), value_max)
+
+class Mavic (Robot):
+    # Constants, empirically found.
+    K_VERTICAL_THRUST = 68.5  # with this thrust, the drone lifts.
+    K_VERTICAL_OFFSET = 0.6   # Vertical offset where the robot actually targets to stabilize itself.
+    K_VERTICAL_P = 3.0        # P constant of the vertical PID.
+    K_ROLL_P = 50.0           # P constant of the roll PID.
+    K_PITCH_P = 30.0          # P constant of the pitch PID.
+
+    MAX_YAW_DISTURBANCE = 0.4
+    MAX_PITCH_DISTURBANCE = -1
+    target_precison = 0.5 #precision between the target position and the robot position in meters
+
+    def __init__(self):
+        Robot.__init__(self)
+
+        self.timeStep = int(self.getBasicTimeStep())
+        
+        self.water_to_drop = 0
+        
+        # Get and enable devices.
+        self.camera = self.getDevice("camera")
+        self.camera.enable(self.timeStep)
+        self.imu = self.getDevice("inertial unit")
+        self.imu.enable(self.timeStep)
+        self.gps = self.getDevice("gps")
+        self.gps.enable(self.timeStep)
+        self.gyro = self.getDevice("gyro")
+        self.gyro.enable(self.timeStep)
+
+        self.front_left_motor = self.getDevice("front left propeller")
+        self.front_right_motor = self.getDevice("front right propeller")
+        self.rear_left_motor = self.getDevice("rear left propeller")
+        self.rear_right_motor = self.getDevice("rear right propeller")
+        self.camera_pitch_motor = self.getDevice("camera pitch")
+        self.camera_pitch_motor.setPosition(1.55) #vertical PoV
+        motors = [self.front_left_motor, self.front_right_motor, self.rear_left_motor, self.rear_right_motor]
+        for motor in motors:
+            motor.setPosition(float('inf'))
+            motor.setVelocity(1)
+
+        self.curr_pos = 6*[0] # X,Y,Z, yaw, pitch, roll
+        self.target_pos = [0,0,0]
```

Comment:
Please do not use abbreviations, they are often confusing as here:
```suggestion
        self.current_pose = 6*[0] # X,Y,Z, yaw, pitch, roll
        self.target_position = [0,0,0]
```